src/getExtendedPublicKey.c

```diff
@@ -66,6 +66,34 @@ void initializePath(uint8_t *dataBuffer)
 	VALIDATE_PARAM(data.bip32Path[2] >= HARDENED_BIP32);
 }
 
+static void os_swap_bigint32(unsigned char *bigint)
+{
+	for (int i = 0; i<16; i++) {
+		unsigned char x;
+		x = bigint[i];
+		bigint[i] = bigint[31-i];
+		bigint[31-i] = x;
+	}
+}
+
+// this is a workaround for SDK <= 1.5.4?, we use this instead of the cx_ecfp_generate_pair
+// as these functions do not work with extended private key
+void derivePublicKey()
```

Comment:
Later we would need to make these functions to get in/out data from parameters instead of a global state